File: src/main/java/com/probielab/microiot/api/router/SockJSRouter.java
package com.probielab.microiot.api.router;

import io.vertx.ext.web.handler.sockjs.SockJSHandlerOptions;
import io.vertx.reactivex.core.Vertx;
import io.vertx.reactivex.ext.web.handler.sockjs.SockJSHandler;

public class SockJSRouter {
  public static SockJSRouter sockJSRouter;
  private SockJSHandler sockJSHandler;

  public static SockJSRouter getInstance(Vertx vertx) {
    if (sockJSRouter == null) {
      sockJSRouter = new SockJSRouter(vertx);
    }
    return sockJSRouter;
  }

  private SockJSRouter(Vertx vertx) {
    SockJSHandlerOptions options = new SockJSHandlerOptions().setHeartbeatInterval(2000);
    sockJSHandler = SockJSHandler.create(vertx, options);
  }

  public SockJSHandler getSockJSHandler() {
    return sockJSHandler;
  }

  private void init(SockJSHandler sh) {
    sh.socketHandler(socket -> {

    });
  }
}

File: src/main/kotlin/com/probielab/microiot/api/router/SockJSRouter.kt
package com.probielab.microiot.api.router

import io.vertx.ext.web.handler.sockjs.SockJSHandlerOptions
import io.vertx.reactivex.core.Vertx
import io.vertx.reactivex.ext.web.handler.sockjs.SockJSHandler

class SockJSRouter private constructor(vertx: Vertx) {
  private val sockJSHandler: SockJSHandler

  private fun init(sh: SockJSHandler) {
    sh.socketHandler { }
  }

  companion object {
    private var sockJSRouter: SockJSRouter? = null
    fun getInstance(vertx: Vertx): SockJSRouter? {
      if (sockJSRouter == null) {
        sockJSRouter = SockJSRouter(vertx)
      }
      return sockJSRouter
    }
  }

  init {
    val options = SockJSHandlerOptions().setHeartbeatInterval(2000)
    sockJSHandler = SockJSHandler.create(vertx, options)
    init(sockJSHandler)
  }
}
